misc/scripts/plugin_sanity_test.sh

```diff
@@ -0,0 +1,122 @@
+
+# To sanity test on the same VM using managed plugin
+# More information https://github.com/vmware/docker-volume-vsphere/issues/1020#issue-213470358
+
+echo "plugin_sanity_test: [INFO] Running plugin_sanity_test on the clean test setup..."
+
+# make sure sanity check is on clean slate and row count should be 2
+row_count=`docker plugin ls | wc -l`
+if [ $row_count != 2 ]; then
+    echo "plugin_sanity_test: [ERROR] error has occurred while counting pluginName row...current count is " $row_count
+    exit 1
+fi
+
+# get installed plugin name
+pluginName=`docker plugin ls | sed 1,1d | awk '{ print $2 }'`
+echo "plugin_sanity_test: [INFO] Installed plugin name is:" $pluginName
+
+# make sure plugin name is not empty
+if [ -z $pluginName ]; then
+    echo "plugin_sanity_test: [ERROR] error has occurred while fetching created plugin name..
+         please make sure plugin is built correctly or not"
+         exit 1
+fi
+
+# check plugin is disabled or not .. by default it should be disabled
+echo "plugin_sanity_test: [INFO] check plugin is disabled or not .. by default it should be disabled..."
+enabled=`docker plugin inspect $pluginName -f {{.Enabled}}`
+if [ $? -ne 0  -o  $enabled != "false" ]; then
+        echo "[ERROR] plugin_sanity_test: docker plugin inspect failed..[" $enabled "]"
+        exit 1
+fi
+
+# setting random value for plugin name
+volumeName="volume_`date +%Y%m%d%H%M%S`";
+
+# create volume when plugin is disabled state and expects an error
+echo "plugin_sanity_test: [INFO] create volume when plugin is disabled state and expects an error..."
+docker volume create -d $pluginName --name $volumeName
+if [ $? -eq 0 ]; then
+        echo "[ERROR] plugin_sanity_test: volume should not be created as plugin is not enabled yet"
+        exit 1
+fi
+
+# Enable plugin and verifies it is enabled or not
+echo "plugin_sanity_test: [INFO] Enable plugin and verifies it is enabled or not..."
+docker plugin enable $pluginName
+enabled=`docker plugin inspect $pluginName -f {{.Enabled}}`
+if [ $? -ne 0  -o  $enabled != "true" ]; then
+        echo "[ERROR] plugin_sanity_test: docker plugin inspect failed..[" $enabled "]"
+        exit 1
+fi
+
+# try to create volume again and expects volume creation this time as plugin is enabled
+echo "plugin_sanity_test: [INFO] try to create volume again and expects volume creation this time as plugin is enabled..."
+docker volume create -d $pluginName --name $volumeName
+if [ $? -ne 0 ]; then
+        echo "[ERROR] plugin_sanity_test: Error occurred while creating volume after plugin enablement"
+        exit 1
+fi
+
+# setting random value for container name
+containerName="container_`date +%Y%m%d%H%M%S`";
+fileName="file_`date +%Y%m%d%H%M%S`";
+
+# run busybox and write something to this volume
+echo "plugin_sanity_test: [INFO] run busybox and write something to this volume"
+docker run --rm -v $volumeName:/vol1 --name=$containerName busybox touch /vol1/$fileName
+if [ $? -ne 0 ]; then
```

Comment:
those 3 lines seem to be everywhere.... I'd consider a functoin for that (this is a nit)